test/rhsmlib/dbus/test_config.py

```diff
@@ -10,64 +10,63 @@
 # Red Hat trademarks are not licensed under GPLv2. No permission is
 # granted to use or replicate Red Hat trademarks that are incorporated
 # in this software or its documentation.
+import tempfile
+from typing import Any, Dict
 
 import dbus
 
 from rhsm.config import RhsmConfigParser
-from rhsmlib.dbus import constants
 from rhsmlib.dbus.objects.config import ConfigDBusObject
-from test.rhsmlib.base import DBusObjectTest, TestUtilsMixin
+from test.rhsmlib.base import DBusServerStubProvider
 
 from test.rhsmlib.services.test_config import TEST_CONFIG
-from test import subman_marker_dbus
 
 
-@subman_marker_dbus
-class TestConfigDBusObject(DBusObjectTest, TestUtilsMixin):
-    def setUp(self):
-        super(TestConfigDBusObject, self).setUp()
-        self.proxy = self.proxy_for(ConfigDBusObject.default_dbus_path)
-        self.interface = dbus.Interface(self.proxy, constants.CONFIG_INTERFACE)
+class TestConfigDBusObject(DBusServerStubProvider):
+    dbus_class = ConfigDBusObject
+    dbus_class_kwargs: Dict[str, Any] = {"parser": None}
 
-    def dbus_objects(self):
-        self.fid = self.write_temp_file(TEST_CONFIG)
-        self.addCleanup(self.fid.close)
-        self.parser = RhsmConfigParser(self.fid.name)
-        return [(ConfigDBusObject, {"parser": self.parser})]
+    config_file = None
+    """Attribute referencing file containing test configuration text."""
 
-    def test_get_all(self):
-        def assertions(*args):
-            result = args[0]
-            self.assertIn("server", result)
+    parser = None
+    """Attribute referencing configuration's parser object."""
 
-        dbus_method_args = [""]
-        self.dbus_request(assertions, self.interface.GetAll, dbus_method_args)
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.config_file = tempfile.NamedTemporaryFile()
+        with open(cls.config_file.name, "w") as handle:
+            handle.write(TEST_CONFIG)
+        cls.parser = RhsmConfigParser(cls.config_file.name)
+        cls.dbus_class_kwargs["parser"] = cls.parser
 
-    def test_get_property(self):
-        def assertions(*args):
-            result = args[0]
-            self.assertIn("server.example.com", result)
+        super().setUpClass()
 
-        dbus_method_args = ["server.hostname", ""]
-        self.dbus_request(assertions, self.interface.Get, dbus_method_args)
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.config_file = None
+        super().tearDownClass()
 
-    def test_get_section(self):
-        def assertions(*args):
-            result = args[0]
-            self.assertIn("hostname", result)
+    def test_GetAll(self):
+        result = self.obj.GetAll.__wrapped__(self.obj, "en_US.UTF-8")
```

Comment:
I see the locale hardcoded here; as I see this string in almost every test (ie those that don't check translated stuff), what about making it a class constant in `DBusServerStubProvider`?

Reply:
Class attribute or global constant -- it's the same for me. My goal is to avoid the copy&pasting of `"en_US.UTF-8"` in almost every test, as typo'ing it in one call is too easy.

Also, those tests are not really testing the language, so English is fine (and it's ok from the previous `""`); because of that, I think having that info not repeated in every English test avoids one "distraction" from the actual goal of each test.